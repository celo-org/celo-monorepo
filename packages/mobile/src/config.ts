import { stringToBoolean } from '@celo/utils/src/parsing'
import Config from 'react-native-config'
import config from 'src/geth/network-config'
import { Testnets } from 'src/web3/testnets'
// if I use @celo/utils breaks all the tests for some reason
// tslint:disable-next-line
import * as secretsFile from '../secrets.json'

// extract secrets from secrets.json
const keyOrUndefined = (file: any, secretsKey: any, attribute: any) => {
  if (secretsKey in file) {
    if (attribute in file[secretsKey]) {
      return file[secretsKey][attribute]
    }
  }
  return undefined
}

export const AVAILABLE_LANGUAGES = [
  { name: 'English', code: 'en-US' },
  { name: 'Español (América Latina)', code: 'es-419' },
]

export const isE2EEnv = Config.IS_E2E || false
export const CELO_VERIFIER_DOWNLOAD_LINK = 'https://celo.org/rewards'
export const CELO_VERIFIER_START_MINING_LINK = 'celo://verifier/start'

export const DEFAULT_COUNTRY = Config.DEFAULT_COUNTRY || null

export const TOS_LINK = 'https://celo.org/user-agreement'
export const FAQ_LINK = 'https://celo.org/faq'

export const CELO_SUPPORT_EMAIL_ADDRESS = 'support@celo.org'

export const BALANCE_OUT_OF_SYNC_THRESHOLD = 5 * 60 // 5 minutes

export const ALERT_BANNER_DURATION = 5000

export const NUMBER_INPUT_MAX_DECIMALS = 2
export const MAX_COMMENT_LENGTH = 70

export const INPUT_DEBOUNCE_TIME = 1000 // milliseconds

export const DEV_SETTINGS_ACTIVE_INITIALLY = stringToBoolean(
  Config.DEV_SETTINGS_ACTIVE_INITIALLY || 'false'
)

export const FIREBASE_ENABLED = stringToBoolean(Config.FIREBASE_ENABLED || 'true')

// We need to fallback to `integration` for testing under jest where
// react-native-config is undefined.
export const DEFAULT_TESTNET: Testnets = Config.DEFAULT_TESTNET || 'integration'
export const BLOCKCHAIN_API_URL = config[DEFAULT_TESTNET].blockchainApiUrl

export const SEGMENT_API_KEY = keyOrUndefined(secretsFile, Config.SECRETS_KEY, 'SEGMENT_API_KEY')
export const FIREBASE_WEB_KEY = keyOrUndefined(secretsFile, Config.SECRETS_KEY, 'FIREBASE_WEB_KEY')

export const SENTRY_URL = keyOrUndefined(secretsFile, Config.SECRETS_KEY, 'SENTRY_URL')

export const PROMOTE_REWARDS_APP = false

export const LOCAL_CURRENCY_SYMBOL = Config.LOCAL_CURRENCY_SYMBOL || null

// The number of seconds before the sender can reclaim the payment.
export const ESCROW_PAYMENT_EXPIRY_SECONDS = 172800 // 2 days

export const SHOW_TESTNET_BANNER = stringToBoolean(Config.SHOW_TESTNET_BANNER || 'false')

// The minimum allowed value for a transaction such as a transfer
export const DOLLAR_TRANSACTION_MIN_AMOUNT = 0.01
export const GOLD_TRANSACTION_MIN_AMOUNT = 0.001
