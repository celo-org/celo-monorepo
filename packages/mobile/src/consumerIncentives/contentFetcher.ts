import firebase from '@react-native-firebase/app'

export interface ContentType {
  [lang: string]: any
}

export const fetchConsumerRewardsContent = () =>
  firebase
    .database()
    .ref('consumerIncentives/content')
    .once('value')
    .then((snapshot) => snapshot.val())
