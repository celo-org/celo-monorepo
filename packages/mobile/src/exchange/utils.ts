import BigNumber from 'bignumber.js'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { convertToContractDecimals } from 'src/tokens/saga'
import { getContractKitAsync } from 'src/web3/contracts'

export async function celoTocUsd(celoAmount: BigNumber) {
  const celoAmountInWei = (
    await convertToContractDecimals(celoAmount, CURRENCY_ENUM.GOLD)
  ).integerValue()

  const contractKit = await getContractKitAsync()
  const exchange = await contractKit.contracts.getExchange()
  const exchangeRate = await exchange.getGoldExchangeRate(celoAmountInWei)
  const cUsdAmount = exchangeRate.multipliedBy(celoAmount)

  return cUsdAmount
}
