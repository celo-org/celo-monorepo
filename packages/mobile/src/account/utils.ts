import { ADDRESS_LENGTH } from 'src/exchange/reducer'

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}
