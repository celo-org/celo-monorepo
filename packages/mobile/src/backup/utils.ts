import { sampleSize } from 'lodash'
import { AsyncStorage } from 'react-native'
import { generateMnemonic, wordlists } from 'react-native-bip39'
import { getKey, setKey } from 'src/utils/keyStore'
import Logger from 'src/utils/Logger'

const TAG = 'Backup/utils'

export async function createQuizWordList(mnemonic: string, language: string | null) {
  const disallowedWordSet = new Set(mnemonic.split(' '))
  const languageWordList = getWordlist(language)
  const wordOptions: string = await generateMnemonic(1000, null, languageWordList)
  const quizWordList = new Set(
    [...wordOptions.split(' ')].filter((word: string) => !disallowedWordSet.has(word))
  )
  return [...quizWordList]
}

export function selectQuizWordOptions(correctWord: string, allWords: string[], numOptions: number) {
  const wordOptions = []
  const correctWordPosition = Math.floor(Math.random() * numOptions)
  const randomWordIndexList = sampleSize([...Array(allWords.length).keys()], numOptions - 1)
  let randomWordIndex: number = 0

  for (let i = 0; i < numOptions; i++) {
    if (i === correctWordPosition) {
      wordOptions.push(correctWord)
      continue
    }

    wordOptions.push(allWords[randomWordIndexList[randomWordIndex]])
    randomWordIndex += 1
  }
  return wordOptions
}

export function getWordlist(language: string | null) {
  if (!language) {
    return wordlists.EN
  }

  switch (language.slice(0, 2)) {
    case 'es': {
      return wordlists.ES
    }
    default: {
      return wordlists.EN
    }
  }
}

// TODO(Rossy) Remove after the next alfa testnet reset
export async function getStoredMnemonic(): Promise<string | null> {
  try {
    Logger.debug(TAG, 'Checking keystore for mnemonic')
    let mnemonic = await getKey('mnemonic')
    if (mnemonic) {
      return mnemonic
    }

    Logger.debug(TAG, 'Mnemonic not found in keystore, checking async storage')
    mnemonic = await AsyncStorage.getItem('mnemonic')
    if (mnemonic) {
      await setKey('mnemonic', mnemonic)
      await AsyncStorage.removeItem('mnemonic')
      return mnemonic
    }

    Logger.error(TAG, 'No mnemonic found')
    return null
  } catch (error) {
    Logger.error(TAG, 'Failed to retrieve mnemonic', error)
    return null
  }
}
