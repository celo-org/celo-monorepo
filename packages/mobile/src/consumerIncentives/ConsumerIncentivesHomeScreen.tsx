import Button, { BtnSizes } from '@celo/react-components/components/Button'
import Touchable from '@celo/react-components/components/Touchable'
import Times from '@celo/react-components/icons/Times'
import colors from '@celo/react-components/styles/colors'
import fontStyles from '@celo/react-components/styles/fonts'
import variables from '@celo/react-components/styles/variables'
import { StackScreenProps } from '@react-navigation/stack'
import React, { Fragment } from 'react'
import { useAsync } from 'react-async-hook'
import { Trans, useTranslation } from 'react-i18next'
import { ActivityIndicator, Image, ScrollView, StyleSheet, Text, View } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import LinearGradient from 'react-native-linear-gradient'
import { SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context'
import { useDispatch } from 'react-redux'
import { showError } from 'src/alert/actions'
import { ErrorMessages } from 'src/app/ErrorMessages'
import { CELO_REWARDS_LINK } from 'src/brandingConfig'
import {
  ConsumerIncentivesData,
  fetchConsumerRewardsContent,
} from 'src/consumerIncentives/contentFetcher'
import { Namespaces } from 'src/i18n'
import { consumerIncentives, leaves } from 'src/images/Images'
import { noHeader } from 'src/navigator/Headers'
import { navigate, navigateBack } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { StackParamList } from 'src/navigator/types'
import useTypedSelector from 'src/redux/useSelector'
import { getContentForCurrentLang } from 'src/utils/contentTranslations'
import Logger from 'src/utils/Logger'

const TAG = 'ConsumerIncentivesHomeScreen'

const useConsumerIncentivesContent = () => {
  const contentResult = useAsync<ConsumerIncentivesData>(fetchConsumerRewardsContent, [])
  let texts
  if (contentResult.result) {
    texts = getContentForCurrentLang(contentResult.result.content)
  }
  return {
    content: texts,
    tiers: contentResult.result?.tiers,
    loading: contentResult.loading,
    error: contentResult.error,
  }
}

type Props = StackScreenProps<StackParamList, Screens.ConsumerIncentivesHomeScreen>
export default function ConsumerIncentivesHomeScreen(props: Props) {
  const { t } = useTranslation(Namespaces.consumerIncentives)
  const userIsVerified = useTypedSelector((state) => state.app.numberVerified)
  const { content, tiers, loading, error } = useConsumerIncentivesContent()
  const insets = useSafeAreaInsets()
  const dispatch = useDispatch()

  if (!loading && error) {
    Logger.error(TAG, 'Error while loading remote texts from Firebase', error)
    dispatch(showError(ErrorMessages.FIREBASE_FETCH_FAILED))
    navigateBack()
    return null
  }

  const onPressCTA = () => {
    if (userIsVerified) {
      navigate(Screens.FiatExchangeOptions, { isCashIn: true })
    } else {
      navigate(Screens.VerificationEducationScreen, { hideOnboardingStep: true })
    }
  }

  const onLearnMore = () => navigate(Screens.WebViewScreen, { uri: CELO_REWARDS_LINK })

  return (
    <ScrollView
      style={styles.container}
      contentContainerStyle={styles.scrollContainer}
      showsVerticalScrollIndicator={false}
    >
      <Image source={leaves} style={styles.backgroundImage} resizeMode={'stretch'} />
      <SafeAreaView edges={['bottom']} style={styles.contentContainer}>
        <Touchable
          style={[styles.closeButton, { marginTop: insets.top + 20 }]}
          onPress={navigateBack}
          borderless={true}
          hitSlop={variables.iconHitslop}
        >
          <Times />
        </Touchable>
        {loading && (
          <ActivityIndicator
            size="large"
            color={colors.greenBrand}
            style={styles.loading}
            testID="ConsumerIncentives/Loading"
          />
        )}
        {content && (
          <>
            <Image source={consumerIncentives} />
            <Text style={styles.title}>{t('title')}</Text>
            <Text style={[styles.body, styles.description]}>{t('description')}</Text>
            {tiers &&
              tiers.map((tier, index) => (
                <Fragment key={`tier${tier.celoReward}`}>
                  {index > 0 && (
                    <LinearGradient
                      start={{ x: 0, y: 0 }}
                      end={{ x: 1, y: 0 }}
                      colors={[colors.light, colors.dark, colors.light]}
                      style={styles.separator}
                    />
                  )}
                  <Text style={[styles.body, styles.tier]}>
                    <Trans
                      i18nKey={'getCeloRewards'}
                      ns={Namespaces.consumerIncentives}
                      tOptions={{ reward: tier.celoReward, minBalance: tier.minBalanceCusd }}
                    >
                      <Text style={styles.bold} />
                      <Text style={styles.bold} />
                    </Trans>
                  </Text>
                </Fragment>
              ))}
            {content.extraSubtitle && <Text style={styles.subtitle}>{content.extraSubtitle}</Text>}
            {content.extraBody && <Text style={styles.body}>{content.extraBody}</Text>}
            {!userIsVerified && (
              <>
                <Text style={styles.subtitle}>{content.unverifiedSubtitle}</Text>
                <Text style={styles.body}>{content.unverifiedBody}</Text>
              </>
            )}
            <View style={styles.buttonContainer}>
              <Button
                size={BtnSizes.FULL}
                text={userIsVerified ? t('addCusd') : t('accountScreen10:confirmNumber')}
                onPress={onPressCTA}
                testID="ConsumerIncentives/CTA"
              />
            </View>
            <TouchableOpacity onPress={onLearnMore} testID="ConsumerIncentives/learnMore">
              <Text style={styles.learnMore}>{t('global:learnMore')}</Text>
            </TouchableOpacity>
          </>
        )}
      </SafeAreaView>
    </ScrollView>
  )
}

ConsumerIncentivesHomeScreen.navOptions = noHeader

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  scrollContainer: {
    flexGrow: 1,
  },
  contentContainer: {
    flex: 1,
    alignItems: 'center',
    marginHorizontal: 24,
  },
  backgroundImage: {
    position: 'absolute',
    width: '100%',
    top: 0,
    left: 0,
    right: 0,
  },
  closeButton: {
    alignSelf: 'flex-start',
  },
  loading: {
    height: '100%',
  },
  title: {
    ...fontStyles.h2,
    marginTop: 16,
    textAlign: 'center',
  },
  description: {
    marginBottom: 36,
    marginTop: 10,
  },
  subtitle: {
    ...fontStyles.h2,
    fontSize: 20,
    marginTop: 40,
    marginBottom: 10,
    textAlign: 'center',
  },
  body: {
    ...fontStyles.regular,
    textAlign: 'center',
  },
  separator: {
    height: 1,
    width: '100%',
    opacity: 0.3,
    marginVertical: 18,
  },
  tier: {
    marginBottom: 4,
    paddingHorizontal: 12,
  },
  bold: {
    fontWeight: 'bold',
  },
  perMonth: {
    ...fontStyles.small,
    color: colors.gray4,
  },
  buttonContainer: {
    marginTop: 36,
    width: '100%',
  },
  learnMore: {
    ...fontStyles.notificationHeadline,
    fontSize: 17,
    alignSelf: 'center',
    marginVertical: 24,
    color: colors.greenUI,
  },
})
