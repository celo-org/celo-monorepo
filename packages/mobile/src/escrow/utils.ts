import { ensureLeading0x } from '@celo/base'
import { generateDeterministicInviteCode } from '@celo/utils/lib/account'
import { publicKeyToAddress } from '@celo/utils/lib/address'

export const generateEscrowPaymentIdAndPk = (
  recipientPhoneHash: string,
  recipientPepper: string,
  addressIndex: number = 0
) => {
  const { publicKey, privateKey } = generateDeterministicInviteCode(
    recipientPhoneHash,
    recipientPepper,
    addressIndex
  )
  return { paymentId: publicKeyToAddress(publicKey), privateKey: ensureLeading0x(privateKey) }
}

export const generateUniquePaymentId = (
  existingPaymentIds: string[],
  phoneHash: string,
  pepper: string
) => {
  const paymentIdSet: Set<string> = new Set(existingPaymentIds)

  // Using an upper bound of 100 to be sure this doesn't run forever given
  // the realistic amount of pending escrow txs is far less than this
  for (let i = 0; i < 100; i += 1) {
    const { paymentId } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
    if (!paymentIdSet.has(paymentId)) {
      return paymentId
    }
  }
}
