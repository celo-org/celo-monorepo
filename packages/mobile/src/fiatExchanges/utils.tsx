import { SIMPLEX_URI } from 'src/config'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { LocalCurrencyCode } from 'src/localCurrency/consts'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { navigateToURI } from 'src/utils/linking'

export const openMoonpay = (currencyCode: LocalCurrencyCode, currencyToBuy: CURRENCY_ENUM) => {
  navigate(Screens.MoonPay, {
    localAmount: 0,
    currencyCode,
    currencyToBuy,
  })
}

export const openSimplex = (account: string | null) => {
  navigateToURI(`${SIMPLEX_URI}?address=${account}`)
}
