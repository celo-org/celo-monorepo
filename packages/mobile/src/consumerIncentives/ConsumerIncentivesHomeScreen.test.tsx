import * as React from 'react'
import 'react-native'
import { fireEvent, render, waitForElement } from 'react-native-testing-library'
import { Provider } from 'react-redux'
import { CELO_REWARDS_LINK } from 'src/config'
import ConsumerIncentivesHomeScreen from 'src/consumerIncentives/ConsumerIncentivesHomeScreen'
import { fetchConsumerRewardsContent } from 'src/consumerIncentives/contentFetcher'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { createMockStore, getMockStackScreenProps } from 'test/utils'

const mockScreenProps = getMockStackScreenProps(Screens.ConsumerIncentivesHomeScreen)

const createStore = (numberVerified: boolean) =>
  createMockStore({
    app: { numberVerified },
  })

jest.mock('src/consumerIncentives/contentFetcher', () => ({
  fetchConsumerRewardsContent: jest.fn(() =>
    Promise.resolve({
      content: {
        en: {
          title: 'Title',
          description: 'Description',
          subtitle1: 'Subtitle 1',
          body1: 'Body 1',
          subtitle2: 'Subtitle 2',
          body2: 'Body 2',
        },
      },
      tiers: [
        { minBalanceCusd: 20, celoReward: 1 },
        { minBalanceCusd: 100, celoReward: 5 },
        { minBalanceCusd: 500, celoReward: 10 },
      ],
    })
  ),
}))

describe('ConsumerIncentivesHomeScreen', () => {
  beforeEach(() => jest.useRealTimers())

  it('renders correctly', async () => {
    const tree = render(
      <Provider store={createStore(true)}>
        <ConsumerIncentivesHomeScreen {...mockScreenProps} />
      </Provider>
    )

    expect(tree.queryByTestId('ConsumerIncentives/Loading')).toBeTruthy()
    expect(tree.queryByTestId('ConsumerIncentives/CTA')).toBeFalsy()

    await waitForElement(() => tree.queryByTestId('ConsumerIncentives/CTA'))

    expect(tree.queryByTestId('ConsumerIncentives/Loading')).toBeFalsy()
    expect(tree.queryByTestId('ConsumerIncentives/CTA')).toBeTruthy()
    expect(fetchConsumerRewardsContent).toHaveBeenCalledTimes(1)
    expect(tree).toMatchSnapshot()
  })

  it('navigates to cash in screen if user is verified and CTA is tapped', async () => {
    const { getByTestId } = render(
      <Provider store={createStore(true)}>
        <ConsumerIncentivesHomeScreen {...mockScreenProps} />
      </Provider>
    )
    await waitForElement(() => getByTestId('ConsumerIncentives/CTA'))

    fireEvent.press(getByTestId('ConsumerIncentives/CTA'))
    expect(navigate).toHaveBeenCalledWith(Screens.FiatExchangeOptions, { isCashIn: true })
  })

  it('navigates to Phone Confirmation screen if user is not verified and CTA is tapped', async () => {
    const { getByTestId } = render(
      <Provider store={createStore(false)}>
        <ConsumerIncentivesHomeScreen {...mockScreenProps} />
      </Provider>
    )
    await waitForElement(() => getByTestId('ConsumerIncentives/CTA'))

    fireEvent.press(getByTestId('ConsumerIncentives/CTA'))

    expect(navigate).toHaveBeenCalledWith(Screens.VerificationEducationScreen, {
      hideOnboardingStep: true,
    })
  })

  it('opens a WebView when Learn More is tapped', async () => {
    const { getByTestId } = render(
      <Provider store={createStore(true)}>
        <ConsumerIncentivesHomeScreen {...mockScreenProps} />
      </Provider>
    )
    await waitForElement(() => getByTestId('ConsumerIncentives/learnMore'))

    fireEvent.press(getByTestId('ConsumerIncentives/learnMore'))

    expect(navigate).toHaveBeenCalledWith(Screens.WebViewScreen, {
      uri: CELO_REWARDS_LINK,
    })
  })
})
