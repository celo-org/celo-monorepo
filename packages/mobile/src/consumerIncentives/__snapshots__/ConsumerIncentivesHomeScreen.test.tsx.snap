// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerIncentivesHomeScreen renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 16,
    }
  }
>
  <View>
    <RNCSafeAreaView
      edges={
        Array [
          "bottom",
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Jost-Medium",
            "fontSize": 22,
            "lineHeight": 28,
            "marginBottom": 8,
            "marginTop": 16,
          }
        }
      >
        Title
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 18,
            "marginTop": 16,
          }
        }
      >
        Description
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Inter-Medium",
            "fontSize": 17,
            "lineHeight": 16,
            "marginTop": 24,
          }
        }
      >
        Subtitle 1
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 18,
            "marginTop": 16,
          }
        }
      >
        Body 1
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Inter-Medium",
            "fontSize": 17,
            "lineHeight": 16,
            "marginTop": 24,
          }
        }
      >
        Subtitle 2
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 18,
            "marginTop": 16,
          }
        }
      >
        Body 2
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
            },
            Object {
              "marginTop": 24,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 100,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            nativeBackgroundAndroid={
              Object {
                "attribute": "selectableItemBackground",
                "rippleRadius": undefined,
                "type": "ThemeAttrAndroid",
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1AB775",
                "flexGrow": 1,
                "height": 48,
                "justifyContent": "center",
                "opacity": undefined,
                "paddingHorizontal": 24,
                "paddingVertical": 5,
              }
            }
            testID="ConsumerIncentives/CTA"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 16,
                  "lineHeight": 22,
                }
              }
            >
              fiatExchangeFlow:addCusd
            </Text>
          </View>
        </View>
      </View>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="ConsumerIncentives/learnMore"
      >
        <View
          accessible={true}
          style={
            Array [
              Object {},
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#1AB775",
                "fontFamily": "Inter-Medium",
                "fontSize": 14,
                "lineHeight": 18,
                "marginTop": 24,
              }
            }
          >
            global:learnMore
          </Text>
        </View>
      </RNGestureHandlerButton>
    </RNCSafeAreaView>
  </View>
</RCTScrollView>
`;
